### DISCUSSION TITLE
Might be a bug. Crons strings (Weekly vs Monthly on weeknumber and day of week)

### DISCUSSION THREAD

**Discussion by s3YwCf2ZbfJG4SHAfjQMAjtsf** [quartznet/quartznet#1857] [General]
I reported a bug with CronExpressionDescriptor for Weekly as shown Quartz.Net vs CED vs cronmaker.com:
https://github.com/bradymholt/cron-expression-descriptor/issues/161

However, I found that 
Weekly Tuesday at a time: 0 10 5 ? * 3 - At 05:10 AM, only on Wednesday <= It should be Tuesday, but 3 instead of TUE is used.

```
            var t = CronScheduleBuilder.WeeklyOnDayAndHourAndMinute(dayOfWeek.Tuesday, 5, 10).Build() as CronTriggerImpl;
            return t.CronExpressionString;
```
The above generates: 0 10 5 ? * 3, instead of with TUE.    

vs: 
Monthly on the second Thursday at a time: 0 45 9 ? 1/1 THU#2 * - At 09:45 AM, on the second Thursday of the month 

Is there an issue in CronScheduleBuilder that it should output the 3 character day of the week instead of the number?  Especially since day of week is 0 based, and the 3 should be 2 for Tuesday.

When I look at the CronExpression on the trigger of job with that cron, it shows as TUE.

**Comment by lahma** [comment]
I don't think there's a bug in the sense of output. For Quartz 3 is Tuesday. But for libraries like cron expression descriptor it would be easier to get output of TUE so you wouldn't need to configure the library to understand where the week starts from.

```csharp
var trigger = (ICronTrigger) TriggerBuilder.Create()
    .WithSchedule(CronScheduleBuilder.WeeklyOnDayAndHourAndMinute(DayOfWeek.Tuesday, 5, 10))
    .Build();

Console.WriteLine($"CRON: {trigger.CronExpressionString}");
Console.WriteLine($"Next fire times: {string.Join(", ", TriggerUtils.ComputeFireTimes((IOperableTrigger) trigger, null, 3))}");
```

Outputs:

```
CRON: 0 10 5 ? * 3
Next fire times: 2022-12-13 03:10:00 +00:00, 2022-12-20 03:10:00 +00:00, 2022-12-27 03:10:00 +00:00
```

So fires will happen on Tuesdays.
